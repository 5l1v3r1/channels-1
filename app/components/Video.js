// @flow
import React from 'react';
import styles from './Video.scss';

const Video = ({
  title,
  thumbnail,
  description,
  publishedAt,
  viewCount
}: {
  title: string,
  thumbnail: string,
  description: string,
  publishedAt: string,
  viewCount: string
}) => (
  <li className={styles.video}>
    <div className={styles.thumbnail}>
      <img src={thumbnail} alt={title} />
    </div>
    <div className={styles.information}>
      <h3 className={`${styles.title} ${styles['css-truncate']} ${styles['css-truncate-target']}`}>{title}</h3>
      <span className={styles.date}>{publishedAt}</span>
      <span className={styles.views}>{viewCount} views</span>
      <p className={styles.description}>{description}</p>
    </div>
  </li>
);

export default Video;
