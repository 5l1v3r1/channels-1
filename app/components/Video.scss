@import '../app.global.scss';

.video {
  display: flex;
  align-items: center;
  height: 200px;
  border-bottom: 1px solid $light-gray;
}

.thumbnail {
  line-height: 0;

  & img {
    width: 250px;
    height: 140px;
    object-fit: cover;
  }
}

.information {
  height: 140px;
  margin: 0 1rem;
  overflow: hidden;
}

.title {
  margin: 0;

  @include font-size(16);
}

.css-truncate-target {
  display: block !important;
  max-width: 100% !important;
}

.date,
.views {
  @include font-size(12);
}

.date {
  margin-right: 1rem;
}

.description {
  margin: 1rem 0 0;
  color: $dark-gray;
  word-wrap: break-word;
}
