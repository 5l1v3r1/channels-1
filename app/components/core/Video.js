// @flow
import React from 'react';
import VideoType from '../../dataTypes/videoType';
import styles from './Video.scss';

const Video = ({
  title,
  thumbnail,
  description,
  publishedAt,
  viewCount,
  handleClick
}: {
  title: string,
  thumbnail: string,
  description?: string,
  publishedAt: string,
  viewCount: string,
  handleClick: (video: VideoType) => void
}) => (
  <li className={description != null ? styles.videoWithDescription : styles.video} onClick={handleClick}>
    <img className={styles.thumbnail} src={thumbnail} alt={title} />
    <div className={styles.information}>
      <h3 className={[styles.title, styles['css-truncate'], styles['css-truncate-target']].join(' ')}>{title}</h3>
      <span className={styles.date}>{publishedAt}</span>
      <span className={styles.views}>{viewCount} views</span>
      {description &&
        <p className={styles.description}>{description}</p>
      }
    </div>
  </li>
);

export default Video;
