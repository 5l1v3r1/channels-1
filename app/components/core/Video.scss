@import '../../app.global.scss';

.video {
  display: flex;
	align-items: center;
	padding: 1rem;
	border-bottom: 1px solid $light-gray;

	&:last-child {
		border-bottom: none;
	}
}

.videoWithDescription {
	@extend .video;
	padding: 2rem 0;

	.thumbnail {
		width: 250px;
		min-width: 250px;
		height: 140px;
	}

	.information {
		height: 140px;
	}

	.title {
		@include font-size(16);
	}
}

.thumbnail {
	width: 125px;
	min-width: 125px;
	height: 70px;
	object-fit: cover;
}

.information {
  margin: 0 1rem;
  overflow: hidden;
}

.title {
  margin: 0;

  @include font-size(14);
}

.css-truncate-target {
  display: block !important;
  max-width: 100% !important;
}

.date,
.views {
	display: inline-block;

  @include font-size(12);
}

.date {
  margin-right: 1rem;
}

.description {
  margin: 1rem 0 0;
  color: $dark-gray;
  word-wrap: break-word;
}
