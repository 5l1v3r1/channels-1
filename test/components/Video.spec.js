import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Video from '../../app/components/core/Video';

function setup() {
  const props = {
    title: 'Professional Advice from Adobe MAX Conference',
    thumbnail: 'https://i.ytimg.com/hqdefault.jpg',
    publishedAt: 'a year ago',
    viewCount: '5,463',
    handleClick: jest.fn()
  };

  const enzymeWrapper = shallow(<Video {...props} />);

  return {
    props,
    enzymeWrapper
  };
}

describe('Video', () => {
  describe('without description', () => {
    it('should render self', () => {
      const { enzymeWrapper } = setup();
      const tree = toJson(enzymeWrapper);

      expect(tree).toMatchSnapshot();
    });
  });

  describe('with description', () => {
    it('should render self', () => {
      const { enzymeWrapper } = setup();
      enzymeWrapper.setProps({
        description: 'I visited Adobe Max and asked all my design heroes for advice.'
      });
      const tree = toJson(enzymeWrapper);

      expect(tree).toMatchSnapshot();
    });
  });

  it('should handle click', () => {
    const { props, enzymeWrapper } = setup();

    enzymeWrapper.simulate('click');

    expect(props.handleClick).toBeCalled();
  });
});
