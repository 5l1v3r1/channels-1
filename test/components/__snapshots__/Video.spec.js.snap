// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video with description should render self 1`] = `
<li
  className="video withDescription"
  onClick={[Function]}
>
  <img
    alt="Professional Advice from Adobe MAX Conference"
    className="thumbnail"
    src="https://i.ytimg.com/hqdefault.jpg"
  />
  <div
    className="information"
  >
    <h3
      className="title css-truncate css-truncate-target"
    >
      Professional Advice from Adobe MAX Conference
    </h3>
    <span
      className="date"
    >
      a year ago
    </span>
    <span
      className="views"
    >
      5,463
       views
    </span>
    <p
      className="description"
    >
      I visited Adobe Max and asked all my design heroes for advice.
    </p>
  </div>
</li>
`;

exports[`Video without description should render self 1`] = `
<li
  className="video"
  onClick={[Function]}
>
  <img
    alt="Professional Advice from Adobe MAX Conference"
    className="thumbnail"
    src="https://i.ytimg.com/hqdefault.jpg"
  />
  <div
    className="information"
  >
    <h3
      className="title css-truncate css-truncate-target"
    >
      Professional Advice from Adobe MAX Conference
    </h3>
    <span
      className="date"
    >
      a year ago
    </span>
    <span
      className="views"
    >
      5,463
       views
    </span>
  </div>
</li>
`;
